utils/swift_build_support/swift_build_support/targets.py

```diff
@@ -12,60 +12,115 @@
 import platform
 
 
-class StdlibDeploymentTarget(object):
+class Platform(object):
+    """
+    Abstract representation of a platform Swift can run on.
+    """
+
+    def __init__(self, name, archs, sdk_name=None, is_darwin=False):
+        """
+        Create a platform with the given name and list of architectures.
+        """
+        self.name = name
+        self.targets = [Target(self, arch) for arch in archs]
+        self.sdk_name = name.upper() if sdk_name is None else sdk_name
```

Comment:
I'm not sure about that. I hate having a bunch of different names for the same thing, so I think it *is* actually a good behavior to default to naming it in a canonical fashion.

This is so that platforms like Linux get a sensible default "sdk_name" (even though that concept doesn't make a ton of sense).